### PR TITLE
Added Jest and some test snapshots.

### PR DESCRIPTION
**Pull request by JamesFTW**
Some test for Freights static components.  For now they are just snapshots.  Can add more comprehensive test later.

### CODE REVIEW

**Comment by benvinegar** [comment]
Sweet! I'll take a look at this tomorrow.

File: static/tests/Layout.test.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      "staging": {
+        "defaultRef": "HEAD"
+      },
+    },
+    "id": "1",
+    "name": "freight",
+    "repository": "https://github.com/getsentry/freight.git"
+  }]
+
+  const children = {
```
Comment (benvinegar) [inline static/tests/Layout.test.js]: Is this a mock `react-router`? Do we need an actual full-blown mock router? Or can we just pass a `<Component/>`?

File: static/tests/Link.test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// Link.react-test.js
+import React from 'react';
+import Link from '../Link';
```
Comment (benvinegar) [inline static/tests/Link.test.js]: I don't see a `Link.jsx`, only `ListLink.jsx`. Where's this coming from?

File: static/tests/RouterNotFound.test.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import RouteNotFound from '../components/RouteNotFound.jsx';
+
+
+test('RouteNotFound Snapshot', () => {
+  const component = renderer.create(
+    <RouteNotFound />
+  );
+  let tree = component.toJSON();
+  expect(tree).toMatchSnapshot();
+})
```
Comment (benvinegar) [inline static/tests/RouterNotFound.test.js]: So, `RouteNotFound` is basically a completely static component.

I don't think it's worthwhile to test these. Basically, if you ever update this file, the snapshot test will say "hey, you changed this – did you expect to?". And of course you did – it's the only possible thing you could have meant (since there's no behavior to change).

I'd remove this test, and any other snapshot tests of pure static components.

File: static/tests/TaskSummary.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+import TaskSummary from '../components/TaskSummary.jsx';
+
+//TODO: figure out why snapshots are rendering null.
+test('TaskSummary Snapshot', () => {
+  const task = {
+    "app": {
+      id: "1",
+      "name": "freight"
+    },
+    "dataCreated": "2017-07-13T18:30:29.941695Z",
```
Comment (benvinegar) [inline static/tests/TaskSummary.test.js]: `dataCreated` or `dateCreated`?

File: static/tests/TooltipTrigger.test.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    },
+    "id": "1",
+    "name": "freight",
+    "repository": "https://github.com/getsentry/freight.git"
+  }]
+
+
+const shallowRenderer = ReactTestUtils.createRenderer();
+const result = shallowRenderer.render(<TooltipTrigger {...props} children={props.title} />);
+expect(result).toMatchSnapshot();
```
Comment (benvinegar) [inline static/tests/TooltipTrigger.test.js]: Indentation in this and other files is inconsistent.

File: static/components/TaskDetails.jsx [hunk 6]
```diff
@@ -8,21 +8,21 @@ import PollingMixin from "../mixins/polling";
...
-  },
-
   render() {
     return (
       <span className="progress" style={{width: this.props.value + '%'}} />
     );
   }
 });
 
+Progress.propTypes = {
```
Comment (kjlundsgaard) [inline static/components/TaskDetails.jsx]: Is this change necessary? It seems like `propTypes` should be an object (as it is above) when using `React.createClass`, while it's used as a property (the change you've made here) on the class when declaring a component with `extends React.Component`

Comment (JamesFTW) [inline static/components/TaskDetails.jsx]: ```Warning: Accessing PropTypes via the main React package is deprecated. Use the prop-types package from npm instead.```
I did that to stop this annoying propTypes warning.  Might not be 100% necessary (yet?), but made my life easier when debugging stuff.
